src/libstd/sys/wasi/mod.rs

```diff
@@ -83,15 +83,18 @@ pub unsafe fn strlen(mut s: *const c_char) -> usize {
 }
 
 pub unsafe fn abort_internal() -> ! {
-    libc::abort()
+    wasi::proc_exit(127)
```

Comment:
One of the main purposes of the wasi target is to interoperate with C code, so for that I think this will want to continue to call `libc::abort` instead of hard-coding the implementation of `libc::abort` since it may eventually run things like libc destructors. (or at least I know other platforms do something in `abort`)

Reply:
WASI does not have signals right now and destructors are executed via signal handling. See [`abort.c`](https://github.com/CraneStation/wasi-libc/blob/320054e84f8f2440def3b1c8700cedb8fd697bf8/libc-top-half/musl/src/exit/abort.c). When signals will be properly introduced to WASI, we can add something like `proc_raise` and interoperability with C libs will not suffer.